File: CPP/392_IsSubsequence.cpp
// 392 - Is Subsequence

bool isSubsequence(string s, string t) {
    if(s.size() == 0) return true;

    int ct = 0;
    for(int i = 0; i < t.size(); i++) {
        if(ct < s.size() && s[ct] == t[i]) {
            ct++;
        }
    }
    return ct == s.size();
}

File: Java/392_IsSubsequence.java
// 392 - Is Subsequence

public boolean isSubsequence(String s, String t) {
    if(s == null || t == null) return false;
    if(s.isEmpty()) return true;

    int ct = 0;
    for(int i = 0; i < t.length(); i++) {
        if(ct < s.length() && s.charAt(ct) == t.charAt(i)) {
            ct++;
        }
    }
    return ct == s.length();
}